tests/Feature/Store/VoucherManagerTest.php

```diff
@@ -90,6 +89,75 @@ public function testThreeColumnsAreVisible()
             ->seeElement("#allocate-vouchers");
     }
 
+    /** @test */
+    public function itCanShowFamilyWarnings()
+    {
+        /**
+         * Create
+         * - a registration
+         * - on a center
+         * - in an area that checks for ID
+         * - for a family
+         * - with kids who havn't been checked for ID.
+         */
+
+        // Make a sponsor
+        $sponsor = factory(Sponsor::class)->create();
+
+        // Set the verification and extended rules to match it.
+        Config::set('arc.verifies_children', [$sponsor->shortcode]);
+        Config::set('arc.extended_sponsors', [$sponsor->shortcode]);
+
+        // Make a Centre in it.
+        $centre =  factory(Centre::class)->create([
+            'sponsor_id' => $sponsor->id,
+        ]);
+
+        // Make a family, with unverified kids.
+        $family = factory(Family::class)->create();
+        $unverifiedKids = factory(Child::class, 3)->states('unverified')->make();
+        $family->children()->saveMany($unverifiedKids);
+
+        // Link it all up to a registration.
+        $registration = factory(Registration::class)->create([
+            'centre_id' => $centre->id,
+            'family_id' => $family->id,
+        ]);
+
+        // Make a CentreUser
+        $ccUser = factory(CentreUser::class)->create([
+            "name" => "CC test user",
+            "email" => "testccuser@example.com",
+            "password" => bcrypt('test_ccuser_pass'),
+        ]);
+        $ccUser->centres()->attach($centre->id, [
+            'homeCentre' => true
+        ]);
+
+        // Navigate to the page as that user.
+        $this->actingAs($ccUser, 'store')
+            ->visit(URL::route('store.registration.voucher-manager', [ 'id' => $registration ]))
+        ;
+
+        // We can see the warnings area
+        $this->seeElement("#family-warning");
+
+        // Fetch the registration in question and check it's notices are present.
+        $registration->fresh();
+        $notices = $registration->getValuation()->getNoticeReasons();
+        $this->assertGreaterThan(0, count($notices));
+
+        // We can see only family warnings
+        foreach ($notices as $notice) {
+            if ($notice["entity"] == "Family") {
+                $this->see($notice["reason"]);
+            } else {
+                $this->dontSee($notice["reason"]);
+            }
+        }
+    }
+
+    /** @test */
```

Comment:
do you need this if the test function name starts with `test`? Why mixing naming conventions?

Reply:
There have been occasions when phpunit has skipped withiut the directive :man_shrugging: at the moment its witchcraft but im filling in places i missed it out.